flink-utils/src/main/java/org/hypertrace/core/flinkutils/avro/RegistryBasedAvroSerde.java

```diff
@@ -66,6 +66,9 @@ public boolean isEndOfStream(T nextElement) {
     try {
       return serializer.serialize(topicName, element);
     } catch (Exception e) {
+      LOGGER.error(
+          "Failed to serialize object of type " + (element != null ? element.getClass().getName()
+              : "[NULL]"), e);
       return new byte[0];
```

Comment:
I don't think we should return empty array here, rather let the exception be propagated and let apps deal with it.

Reply:
fixed